joplin/js/CreateContentModal/index.js

```diff
@@ -52,70 +52,87 @@ class CreateContentModal extends Component {
+      redirectUrl: dataObj.redirectUrl
     });
     this.incrementActiveStep();
-  }
+  };
 
-  handleTitleInputChange = (e) => {
+  handleTitleInputChange = e => {
     this.setState({
       title: e.target.value,
-      titleCharacterCount: e.target.value.length,
+      titleCharacterCount: e.target.value.length
     });
-  }
+  };
 
-  handleTopicSelect = (id) => {
+  handleTopicSelect = id => {
     this.setState({ topic: id });
-  }
+  };
 
   redirectToEditPage = () => {
     this.writeToLocalStorage();
     window.location.href = this.state.redirectUrl;
-  }
+  };
 
   writeToLocalStorage = () => {
     localStorage.wagtailCreateModal = JSON.stringify(this.state);
-  }
+  };
 
+  handleCloseButton = e => {
+    this.setState({
+      type: null,
+      title: "", // React warning said: `value` prop on `input` should not be null. Consider using an empty string...
+      topic: null,
+      activeStep: 0,
+      redirectUrl: null,
+      titleCharacterCount: 0
+    });
+  };
 
   render() {
     return (
-      <div className="modal fade" id="createNewContentModal" tabIndex="-1" role="dialog" aria-labelledby="createNewContentModalLabel">
+      <div
+        className="modal fade"
+        id="createNewContentModal"
+        tabIndex="-1"
+        role="dialog"
+        aria-labelledby="createNewContentModalLabel"
+      >
         <div className="CreateContentModal__wrapper">
           <div className="modal-dialog" role="document">
             <div className="modal-content CreateContentModal">
               <div className="modal-body">
-                { this.state.activeStep === 0 &&
-                  <ChooseTypeStep handleTypeSelect={this.handleTypeSelect}/>
-                }
-                { this.state.activeStep === 1 &&
+                {this.state.activeStep === 0 && (
+                  <ChooseTypeStep handleTypeSelect={this.handleTypeSelect} />
+                )}
+                {this.state.activeStep === 1 && (
                   <ChooseTitleStep
                     pageType={this.state.type}
                     title={this.state.title}
                     handleTitleInputChange={this.handleTitleInputChange}
                     characterCount={this.state.titleCharacterCount}
                     maxCharacterCount={MAX_TITLE_LENGTH}
                   />
-                }
-                { this.state.activeStep === 2 &&
+                )}
+                {this.state.activeStep === 2 && (
                   <ChooseTopicStep
                     topic={this.state.topic}
                     handleTopicSelect={this.handleTopicSelect}
                     themeTopicTree={THEME_TOPIC_TREE}
                   />
-                }
+                )}
                 <ButtonBar
                   handleBackButton={this.handleBackButton}
                   handleNextButton={this.handleNextButton}
+                  handleCloseButton={this.handleCloseButton}
                   activeStep={this.state.activeStep}
```

Comment:
Real big nit, so feel free to ignore. It seems like the only thing that activeStep is used for in ButtonBar is to show or hide the component so we could make this more obvious.
```
<ButtonBar
  handleBackButton={this.handleBackButton}
  handleNextButton={this.handleNextButton}
  handleCloseButton={this.handleCloseButton}
  hideButtonBar={!this.state.activeStep}
/>
```